static/app/views/explore/charts/index.tsx

```diff
@@ -113,6 +116,15 @@ export function ExploreCharts({query}: ExploreChartsProps) {
     [visualizes, setVisualizes]
   );
 
+  // Synchronize chart cursors
+  const [_, setRenderTrigger] = useState(0);
+  useEffect(() => {
+    if (!timeSeriesResult.isPending) {
+      echarts?.connect(EXPLORE_CHART_GROUP);
+      setRenderTrigger(prev => (prev + 1) % Number.MAX_SAFE_INTEGER);
+    }
+  }, [visualizes, timeSeriesResult.isPending]);
```

Comment:
This seems like a hack to re-run `echarts.connect` on every render. Can this be limited to just `[timeseriesResult.data]` so it reconnects each time the timeseries is fetched?

Reply:
Hey, tried `timeseriesResult.data`, the cursors don't persist on hover.